specification/ingest/_types/Database.ts

```diff
@@ -19,15 +19,21 @@
 
 import { Id, Name } from '@_types/common'
 
-export class DatabaseConfiguration {
+specification / ingest / _types / Database.ts
+
+/** The configuration necessary to identify which IP geolocation provider to use to download the database, as well as any provider-specific configuration necessary for such downloading.
+ * At present, the only supported providers are maxmind and ipinfo, and the maxmind provider requires that an account_id (string) is configured.
+ * A provider (either maxmind or ipinfo) must be specified.
+ */
+export type DatabaseConfiguration = Maxmind | Ipinfo
+
+export class DatabaseConfigurationBase {
   /** The provider-assigned name of the IP geolocation database to download. */
   name: Name
-  /** The configuration necessary to identify which IP geolocation provider to use to download the database, as well as any provider-specific configuration necessary for such downloading.
-   * At present, the only supported provider is maxmind, and the maxmind provider requires that an account_id (string) is configured.
-   */
-  maxmind: Maxmind
```

Comment:
We do need the `maxmind` key which is gone with this approach.

Also it's OK to have some duplication, eg I'm happy to keep the previous approach but simply add an `ipinfo` field and type, with one less field.

Reply:
Agreed, yeah, I'll take a swing at fixing that -- it wasn't an intentional change in this second version of things, I just wasn't thoughtful enough about how I did it.

Reply:
What do you think about 31fd86a38351c91a75cd67a0f05f1e0b93ee007c?

Reply:
This looks great, thanks! Can you please confirm that the GeoIP APIs also accept `ipinfo` databases? I'm asking as `DatabaseConfiguration` is currently shared between the two APIs.

Reply:
Yes, they do.